spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/secretmanager/SecretManagerConfigDataLocationResolver.java

```diff
@@ -94,10 +104,21 @@ private static GcpSecretManagerProperties getSecretManagerProperties(
 
   private static GcpProjectIdProvider createProjectIdProvider(
       ConfigDataLocationResolverContext context) {
-    GcpSecretManagerProperties properties = context.getBootstrapContext()
-        .get(GcpSecretManagerProperties.class);
-    return properties.getProjectId() != null
-        ? properties::getProjectId : new DefaultGcpProjectIdProvider();
+
+    ConfigurableBootstrapContext bootstrapContext = context.getBootstrapContext();
+    
+    GcpSecretManagerProperties secretManagerProperties = bootstrapContext.get(GcpSecretManagerProperties.class);
+    if(secretManagerProperties.getProjectId() != null) {
+        return secretManagerProperties::getProjectId;
+    }
+    
+    GcpProperties gcpProperties = bootstrapContext.get(GcpProperties.class);
+    if(gcpProperties.getProjectId() != null) {
+        return gcpProperties::getProjectId;
+    }
```

Comment:
As written in https://github.com/GoogleCloudPlatform/spring-cloud-gcp/issues/3181#issuecomment-2488507623 this needs a little change:
```suggestion
    if (secretManagerProperties.getProjectId() != null) {
      return secretManagerProperties::getProjectId;
    }
    
    GcpProperties gcpProperties = bootstrapContext.get(GcpProperties.class);
    if (gcpProperties.getProjectId() != null) {
      return gcpProperties::getProjectId;
    }
```